scrapy/extensions/logstats.py

```diff
@@ -1,58 +1,134 @@
 import logging
+from datetime import datetime
 
 from twisted.internet import task
 
 from scrapy import signals
 from scrapy.exceptions import NotConfigured
+from scrapy.utils.serialize import ScrapyJSONEncoder
 
 logger = logging.getLogger(__name__)
 
 
 class LogStats:
     """Log basic scraping stats periodically"""
 
-    def __init__(self, stats, interval=60.0):
+    def __init__(
+            self, stats, interval=60.0, extended=False, ext_include=None, ext_exclude=None
+    ):
         self.stats = stats
         self.interval = interval
         self.multiplier = 60.0 / self.interval
         self.task = None
+        self.extended = extended
+        if self.extended:
+            self.encoder = ScrapyJSONEncoder(sort_keys=True, indent=4)
+        self.ext_include = ext_include
+        self.ext_exclude = ext_exclude
 
     @classmethod
     def from_crawler(cls, crawler):
         interval = crawler.settings.getfloat("LOGSTATS_INTERVAL")
+        extended = crawler.settings.getbool("LOGSTATS_EXTENDED_ENABLED")
```

Comment:
Assuming we may eventually get a non-delta alternative implementation to address https://github.com/scrapy/scrapy/issues/2173, I think it may be worth considering to replace the boolean `LOGSTATS_EXTENDED_ENABLED` with a string `LOGSTATS_MODE` that supports `"simple"` (default) and `"delta"`, and in the future also `"stats"`.

We can then remove `_EXTENDED` from the names of the other 2 new settings.